src/utils/handleQuery.ts

```diff
@@ -0,0 +1,9 @@
+import { Todo } from '../types/Todo';
+
+export function handleQuery(todos: Todo[], query: string): Todo[] {
+  return todos.filter(todo => {
+    const normalizedQuery = query.trim().toLowerCase();
```

Comment:
You are normalizing the query on each iteration of the filter method. This is inefficient as the query string remains the same for each todo item. You should move the normalization of the query string outside of the filter method to improve performance.